Add files via upload

JamileQueiroz opened the pull request.

JamileQueiroz committed:
Add files via upload

admshumar commented:
Cool! Now we're more comfortable with performing pull requests. As you can see, we can comment on the content of a pull request. This is how development teams modify and add to their software.

admshumar commented:
Next, we can discuss a couple of ideas for additions to the R code for this project. One important task would be to determine how to get R to automatically generate graphs of the distribution of IDCN members by education for a particular job sector.

<center><img width="586" alt="screen shot 2018-07-23 at 10 22 18 am" src="https://user-images.githubusercontent.com/31727606/43065548-79d31bb2-8e62-11e8-8e10-31a39823a9d7.png"></center>

There are many job sectors, so producing these graphs by hand would be annoying. Thus we want to automate this process.

The code for producing these graphs is found in the file constructGraphs.R. What I think we need to do is the following: we need to write a loop over all of the job sectors and use a graphics driver to write the graph to a .pdf file.

This is not an easy task. However, once you write the proper code and test it, you will have made a substantial contribution to the project!

I expect that we'll be talking a lot about graphics drivers in R, as well as how to properly loop over job events. I'll look for questions on Slack, and we can certainly chat about this after our first team meeting.

JamileQueiroz commented:
About the loop over all of the job sectors, I repeated your code:

print(jobreport_fieldplot(A1, "Research & Development"))
print(jobreport_fieldplot(A1, "Project Management"))
print(jobreport_fieldplot(A1, "Education / Training"))
print(jobreport_fieldplot(A1, "Communications"))
print(jobreport_fieldplot(A1, "Sales / Customer Service / Business Development"))
print(jobreport_fieldplot(A1, "General Management"))
print(jobreport_fieldplot(A1, "Human Resources"))
print(jobreport_fieldplot(A1, "Other"))
print(jobreport_fieldplot(A1, "Marketing / Advertising"))
print(jobreport_fieldplot(A1, "IT / Engineering"))
print(jobreport_fieldplot(A1, "Operations / Logistics"))
print(jobreport_fieldplot(A1, "Accounting/Finance"))
print(jobreport_fieldplot(A1, "Legal / Corporate Affairs / Compliance"))
print(jobreport_fieldplot(A1, "Operations / Logistics"))

...Now, I am trying to replace these repeats with a for loop function:

x <- c("Research & Development","Project Management","Education / Training",
  "Communications","Sales / Customer Service / Business Development",
  "General Management", "Human Resources","Marketing / Advertising", 
  "IT / Engineering", "Operations / Logistics", 
  "Legal / Corporate Affairs / Compliance", "Operations / Logistics")
for (i in x){
  print(jobreport_fieldplot(A1, i))
}

JamileQueiroz commented:
I have completed this code:

`x <- c("Research & Development","Project Management","Education / Training",
  "Communications","Sales / Customer Service / Business Development",
  "General Management", "Human Resources","Marketing / Advertising", 
  "IT / Engineering", "Operations / Logistics", 
  "Legal / Corporate Affairs / Compliance", "Operations / Logistics")

pdf("figure.pdf", width = 12.44, height = 7)
for (i in x){
  print(jobreport_fieldplot(A1, i))
}
dev.off())`

[figure.pdf](https://github.com/admshumar/IDCN-talent-management-platform/files/2240647/figure.pdf)

admshumar commented:
This looks like a solution! It's nice that all of the graphs are together in a single PDF file. Another option is to get R to write each graph to a seperate PDF file, so that we can choose which of these graphs appear in various reports :-)

After we settle on a piece of code, you can pull request your changes in constructGraphs.R, I'll test them, and then we can merge it!

JamileQueiroz committed:
exportGraphs.R

Each graph to a separated PDF file and the graphs together in a single PDF file.

JamileQueiroz committed:
Merge pull request #1 from JamileQueiroz/JamileQueiroz-patch-1

exportGraphs.R

JQSousa committed:
the update the graphs together in a single PDF file and the other situation that each graph to a separate PDF file.I included these alterations in the exportGraphic.

JQSousa committed:
Up-date 12 August 2018:

Insert the XLS package
Create the TMP_active_namesmails.xlsx

JQSousa committed:
Error

Merge branch 'master' of https://github.com/JamileQueiroz/IDCN-talent-management-platform

# Conflicts:
#	.gitignore
#	TMP-R/.DS_Store
#	TMP-R/MonthlyReport.R
#	TMP-R/loadLibraries.R

JamileQueiroz committed:
Update

JamileQueiroz committed:
Update

JamileQueiroz committed:
Update
